src/library_manager/lib_manager.c

```diff
@@ -47,6 +51,71 @@ struct lib_manager_dma_ext {
 
 static struct ext_library loader_ext_lib;
 
+#if CONFIG_LIBRARY_AUTH_SUPPORT
+static int lib_manager_auth_init(void)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+	int ret;
+
+	if (auth_api_version().major != AUTH_API_VERSION_MAJOR)
+		return -EINVAL;
+
+	ext_lib->auth_buffer = rballoc_align(0, SOF_MEM_CAPS_RAM,
+					     AUTH_SCRATCH_BUFF_SZ, CONFIG_MM_DRV_PAGE_SIZE);
+	if (!ext_lib->auth_buffer)
+		return -ENOMEM;
+
+	ret = auth_api_init(&ext_lib->auth_ctx, ext_lib->auth_buffer,
+			    AUTH_SCRATCH_BUFF_SZ, IMG_TYPE_LIB);
+	if (ret != 0) {
+		tr_err(&lib_manager_tr, "lib_manager_auth_init() failed with error: %d", ret);
+		ret = -EACCES;
+	}
+
+	return ret;
+}
+
+static void lib_manager_auth_deinit(void)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+
+	if (ext_lib->auth_buffer)
+		rfree(ext_lib->auth_buffer);
+	ext_lib->auth_buffer = NULL;
+	memset(&ext_lib->auth_ctx, 0, sizeof(struct auth_api_ctx));
+}
+
+static int lib_manager_auth_proc(const void *buffer_data,
+				 size_t buffer_size, enum auth_phase phase)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+	int ret;
+
+	ret = auth_api_init_auth_proc(&ext_lib->auth_ctx, buffer_data, buffer_size, phase);
+
+	if (ret != 0) {
+		tr_err(&lib_manager_tr, "lib_manager_auth_proc() failed with error: %d", ret);
+		return -EACCES;
+	}
+
+	/* The auth_api_busy() will timeouts internally in case of failure */
+	while (auth_api_busy(&ext_lib->auth_ctx))
+		;
+
+	ret = auth_api_result(&ext_lib->auth_ctx);
+
+	if (ret != AUTH_IMAGE_TRUSTED) {
+		tr_err(&lib_manager_tr, "lib_manager_auth_proc() Untrasted library!");
+		return -ENOTSUP;
```

Comment:
not sure which error code would be best here, maybe consider `EACCES`, `EPERM`, `ENOEXEC`, `EILSEQ`

Reply:
fixed